### PR TITLE
c_kzg_4844.c: In verify_blog_kzg_proof_batch, reuse temporary variable.

### PR DESCRIPTION
**Pull request by GottfriedHerold**
Polynomial is only used as a local temporary inside the loop.

There is no need to allocate an array of them.

See #234 

Note that this PR declares the variable inside the loop for clarity.

### CODE REVIEW

File: src/c_kzg_4844.c [hunk 1]
```diff
@@ -1513,16 +1510,18 @@ C_KZG_RET verify_blob_kzg_proof_batch(
         );
         if (ret != C_KZG_OK) goto out;
 
+        Polynomial polynomial;
```
Comment (asn-d6) [inline src/c_kzg_4844.c:1513]: Nitpick: I think we tend to declare variables in the beginning of each scope (that is, right after the `for` line in this case). No strong opinion from my side.

**Comment by asn-d6** [comment]
Looks great! Left a small nit that can also be ignored.

File: src/c_kzg_4844.c (hunk 1 above)
Comment (jtraglia) [reply src/c_kzg_4844.c:1513]: Yeah, I agree with @asn-d6 here. Let's put that at the top of the loop, before `bytes_to_kzg_commitment`.

Comment (GottfriedHerold) [reply src/c_kzg_4844.c:1513]: No strong opinion here, did this mostly to convince myself that (re-)using the variable was safe (this way, further using it after the loop would not compile).

Comment (GottfriedHerold) [reply src/c_kzg_4844.c]: Fixed.

**Review by asn-d6 — APPROVED** [review]
LGTM!

**Pull request merged**